elasticsearch-model/lib/elasticsearch/model/indexing.rb

```diff
@@ -153,7 +153,24 @@ def mapping(options={}, &block)
         #
         #     # => {:index=>{:number_of_shards=>1}}
         #
+        # You can specify a YAML file with settings
+        #
+        # @example Define index settings from YAML file
+        #
+        #     # config/elasticsearch/articles.yml:
+        #     #
+        #     # index:
+        #     #   number_of_shards: 1
+        #     #
+        #
+        #     Article.settings "config/elasticsearch/articles.yml"
+        #
+        #     Article.settings.to_hash
+        #
+        #     # => { "index" => { "number_of_shards" => 1 } }
+        #
         def settings(settings={}, &block)
+          settings = YAML.load_file(settings) if settings.is_a?(String)
```

Comment:
I have an example here: https://github.com/elastic/elasticsearch-rails/blob/master/elasticsearch-persistence/examples/music/index_manager.rb, where the `IndexManager` class simply gathers and merges settings/mappings from a model, and creates an index.

I feel like this is something best left to the application developer, simply because there are dozens of ways how to do this (simply storing a Hash in some global module, having a module method, putting it in YAML and reading it, using an existing gem for that, ...) -- *but* all that said I like the feature you propose. Let's first have that, document it, maybe use it in our [example applications](https://github.com/elastic/elasticsearch-rails/tree/master/elasticsearch-rails/lib/rails/templates) and then discuss if we can put some more sugar on it?